client/post-editor/post-editor.jsx

```diff
@@ -650,14 +649,12 @@ export const PostEditor = React.createClass( {
 
 	onPublish: function() {
 		const edits = {
-			...this.props.edits,
-			status: 'publish'
+			status: 'publish',
+			...this.props.edits
 		};
 
-		// determine if this is a private publish
-		if ( utils.isPrivate( this.state.post ) ) {
-			edits.status = 'private';
-		} else if ( utils.isFutureDated( this.state.post ) ) {
+		// determine if this is a future publish
+		if ( utils.isFutureDated( this.state.post ) ) {
```

Comment:
> With "After" case, we could do similar with future dating to apply it as an edit before calling onPublish.

Yeah! I thought about this! this makes it more explicit instead of relying on the server. seems like `setPostDate` is a good candidate for this logic